ide/xml.text/src/org/netbeans/modules/xml/text/navigator/NavigatorTreeCellRenderer.java

```diff
@@ -91,12 +91,14 @@ public void setIcon(Icon[] icons, boolean containsError) {
         renderer.setIcon(icons[containsError ? 1 : 0]);
     }
     
-    private ImageIcon getImageIcon(String name, boolean error){
-        ImageIcon icon = ImageUtilities.loadImageIcon(name, false);
-        if(error)
-            return new ImageIcon(ImageUtilities.mergeImages( icon.getImage(), ERROR_IMAGE, 15, 7 ));
-        else
+    private Icon getIcon(String name, boolean error){
+        Icon icon = ImageUtilities.loadImageIcon(name, false);
+        if (error) {
+            return ImageUtilities.image2Icon(ImageUtilities.mergeImages(
+                    ImageUtilities.icon2Image(icon), ERROR_IMAGE, 15, 7 ));
+        } else {
             return icon;
+        }
```

Comment:
one call less if the variable keeps the original `ImageIcon` type.
```java
    private Icon getIcon(String name, boolean error){
        ImageIcon icon = ImageUtilities.loadImageIcon(name, false);
        if (error) {
            return ImageUtilities.image2Icon(ImageUtilities.mergeImages(icon.getImage(), ERROR_IMAGE, 15, 7 ));
        } else {
            return icon;
        }
    }
```

Reply:
It's another change I think can be dropped overall since mergeImages is smart enough to preserve HiDPI resolution.